### PR TITLE
NSA-28 - Reset Password - Code functionality

### PR DESCRIPTION
**Pull request by dashton82**
From the story - note that there is no expiry, also this current design doesn't cater for expect different types of codes stored against a user - this will happen in further iterations once other code types have been defined.

### CODE REVIEW

File: src/userCodes/index.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+const validateCode = require('./validate');
+
+const router = express.Router();
+
+const routeExport = () => {
+  // Add auth middleware.
+  router.use(apiAuth(router, config));
+
+  // Map routed to functions.
+  router.post('/create', postCreateCode);
```
Comment (simonbu11) [inline src/userCodes/index.js]: Should this be a put as a user can only have 1 code?

File: src/userCodes/postCreateCode.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+'use strict';
+const logger = require('../logger');
+const redisUserCodeStorage = require('./redisUserCodeStorage');
+
+const post = async (req,res) => {
+
+  try{
+    if(req.body.uid === null || req.body.uid === undefined || req.body.uid === '') {
```
Comment (simonbu11) [inline src/userCodes/postCreateCode.js]: Could shorten to:
```
it(!req.body.uid) {
```

File: src/userCodes/postCreateCode.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      res.status(400).send();
+      return;
+    }
+    const uid = req.body.uid;
+    const storage = new redisUserCodeStorage();
+
+    const code = await storage.getUserPasswordResetCode(uid);
+
+    if(code === null || code === undefined){
```
Comment (simonbu11) [inline src/userCodes/postCreateCode.js]: Can be shorten as above

File: src/userCodes/redisUserCodeStorage.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
+'use strict'
+
+const redis = require('ioredis');
+const config = require('./../config');
+const resetCode = require('./generateCode');
+let client;
+
+class RedisUserCodeStorage {
+  constructor(redisClient){
+    if(redisClient === null || redisClient === undefined){
```
Comment (simonbu11) [inline src/userCodes/redisUserCodeStorage.js]: Can be shorten as above

File: src/userCodes/validate.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
+'use strict';
+const logger = require('../logger');
+const redisUserCodeStorage = require('./redisUserCodeStorage');
+
+const validate = async (req,res) => {
+  try{
+    if(req.params.uid === null || req.params.uid === undefined || req.params.uid === ''
```
Comment (simonbu11) [inline src/userCodes/validate.js]: Can be shorten as above

File: src/userCodes/postCreateCode.js (hunk 2 above)
Comment (floydprice) [reply src/userCodes/postCreateCode.js]: Will still need 
```
req.body.uid === ''
```

File: src/userCodes/index.js (hunk 1 above)
Comment (floydprice) [reply src/userCodes/index.js]: We didn't state that a user can only have one code... Needs a discussion.

**Pull request merged**